ads/google/a4a/utils.js

```diff
@@ -194,17 +195,29 @@ export function googleBlockParameters(a4a, opt_experimentIds) {
  * @return {!Promise<!Object<string,!Array<!Promise<!../../../src/base-element.BaseElement>>>>}
  */
 export function groupAmpAdsByType(win, type, groupFn) {
+  // Look for amp-ad elements of correct type or those contained within
+  // standard container type.  Note that display none containers will not be
+  // included as they will never be measured.
+  // TODO(keithwrightbos): what about slots that become measured due to removal
+  // of display none (e.g. user resizes viewport and media selector makes
+  // visible).
   return Services.resourcesForDoc(win.document).getMeasuredResources(win,
-      r => r.element.tagName == 'AMP-AD' &&
-        r.element.getAttribute('type') == type)
-      .then(resources => {
-        const result = {};
-        resources.forEach(r => {
-          const groupId = groupFn(r.element);
-          (result[groupId] || (result[groupId] = [])).push(r.element.getImpl());
-        });
+      r => (Object.keys(ValidAdContainerTypes).includes(r.element.tagName) &&
```

Comment:
Nit: I know you enjoy your code golf, but maybe we can avoid inlining the function here? SRA code is confusing enough. Something like:

```javascript
r => {
  const inValidContainer = Object.keys(validAdContainerTypes).includes(r.element.tagName) &&
    !!r.element.querySelector(`amp-ad[type=${type}]`));
  const correctType = r.element.tagName == 'AMP-AD' && r.element.getAttribute('type') == type;
  return inValidContainer || correctType;
}

Reply:
![image](https://user-images.githubusercontent.com/4276992/39581629-0306d966-4eba-11e8-92f3-f81b7ed571b3.png)

Reply:
Done... but let my code golf score be known to all!